### PR TITLE
PIN-4122: added '/purposes/eservices/{eserviceId}/consumer/{consumerId}'

### PR DESCRIPTION
**Pull request by AsterITA**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -159,6 +159,132 @@ paths:
               schema:
                 type: integer
               description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+  '/purposes/eservices/{eserviceId}/consumer/{consumerId}':
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: not sure if there are specific requirements, but what about
```suggestion
  '/purposes':
```
 moving `eserviceId` and `consumerId` as optional query parameters?

Comment (AsterITA) [reply src/main/resources/interface-specification.yml]: > not sure if there are specific requirements, but what about
> 
> moving `eserviceId` and `consumerId` as optional query parameters?

I agree on passing those as query parameters but, as I understand, they have to be required

Comment (galales) [reply src/main/resources/interface-specification.yml]: fine by me, the lift of this constraint in the future is not a real breaking change so I guess can be done when required.

Leaving to @beetlecrunch the last word about the change

Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml]: Nothing against the change proposed.
As @AsterITA mentioned, these query parameters must be required.

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -159,6 +159,132 @@ paths:
               schema:
                 type: integer
               description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+  '/purposes':
+    get:
+      tags:
+        - gateway
+      summary: Get purposes for eserviceId/consumerId
+      description: Retrieve all the active purposes associated to eserviceId/consumerId
+      operationId: getPurposesByEserviceAndConsumer
```
Comment (beetlecrunch) [inline src/main/resources/interface-specification.yml]:
```suggestion
      operationId: getPurposes
```

**Review by beetlecrunch — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**